tokio/src/runtime/io/driver.rs

```diff
@@ -196,35 +174,23 @@ impl Driver {
             } else if token == TOKEN_SIGNAL {
                 self.signal_ready = true;
             } else {
-                Self::dispatch(
-                    &mut self.resources,
-                    self.tick,
-                    token,
-                    Ready::from_mio(event),
-                );
-                ready_count += 1;
-            }
-        }
-
-        handle.metrics.incr_ready_count_by(ready_count);
-    }
-
-    fn dispatch(resources: &mut Slab<ScheduledIo>, tick: u8, token: mio::Token, ready: Ready) {
-        let addr = slab::Address::from_usize(ADDRESS.unpack(token.0));
+                let ready = Ready::from_mio(event);
+                let ptr: *const ScheduledIo = token.0 as *const _;
```

Comment:
Miri probably hates this...it could, at least, be worth adding a comment about using [`ptr::from_exposed_addr`](https://doc.rust-lang.org/nightly/std/ptr/fn.from_exposed_addr.html) if and when it becomes stable?